kafka/containers/0.1.0/kafka/run.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# print environment for debugging
+env
+
+# some defaults
+KAFKA_HOME=${KAFKA_HOME:-/opt/kafka}
+KAFKA_CONFIG_FILE=$KAFKA_HOME/config/server.properties
+
+broker_id_config="$KAFKA_HOME/conf.d/broker_id"
+while [ ! -s $broker_id_config ]; do
+  echo "Waiting for $broker_id_config to appear"
+  sleep 1
+done
+broker_id=$(cat $broker_id_config)
+echo "broker.id = $broker_id" > $KAFKA_CONFIG_FILE
+
+zookeeper_connect_config="$KAFKA_HOME/conf.d/zookeeper_connect"
+while [ ! -s $zookeeper_connect_config ]; do
+  echo "Waiting for $zookeeper_connect_config to appear"
+  sleep 1
+done
+zookeeper_connect=$(cat $zookeeper_connect_config | xargs | tr ' ' ',')
+echo "zookeeper.connect = $zookeeper_connect" >> $KAFKA_CONFIG_FILE
+
+export KAFKA_CONFIG_PORT=${KAFKA_CONFIG_PORT:-9092}
+
+if [ -z "$KAFKA_CONFIG_LOG_DIRS" ]; then
+  echo "Must export KAFKA_CONFIG_LOG_DIRS environment variable" >&2
+  exit 1
+else
+  # make sure they are comma separated
+  export KAFKA_CONFIG_LOG_DIRS=$(echo "$KAFKA_CONFIG_LOG_DIRS" | tr ' ' ',')
+fi
+
+export KAFKA_CONFIG_HOST_NAME=${KAFKA_CONFIG_HOST_NAME:-$(hostname)}
+
+for v in `env | egrep '^KAFKA_CONFIG_'`; do
+  name=$(echo "$v" | cut -d '=' -f 1 | sed -e 's/KAFKA_CONFIG_//' | tr '[A-Z]' '[a-z]' | tr '_' '.')
+  value=$(echo "$v" | cut -d '=' -f 2)
+  echo "$name = $value" >> $KAFKA_CONFIG_FILE
+done
+
+echo "Starting kafka .."
+$KAFKA_HOME/bin/kafka-server-start.sh $KAFKA_CONFIG_FILE &
+pid=$!
+
+# wait for kafka to start up
+while ! netstat --listen --numeric --tcp | awk '{ print $4 }' | egrep -q ":$KAFKA_CONFIG_PORT$"; do
+  # make sure we do not wait forever though
+  kill -0 $pid &>/dev/null || { echo "Kafka shut down unexpectedly"; exit 1; }
+  echo "Waiting for kafka to bind to TCP $KAFKA_CONFIG_PORT"
+  sleep 1
+done
+
+if [ -n "$KAFKA_CREATE_TOPICS" ]; then
```

Comment:
Can this logic be in its own sidekick? Possibly define the topics from metadata on the service? 

```
rancher-compose.yml:
  kafka:
    scale: 3
    metadata:
        topics:
           - name:
             partition:
             replica: